ghost_code.py

```diff
@@ -280,15 +297,22 @@ def sprinkle_loop_invariants(func: nip.Function) -> Iterable[Insertion]:
 def sprinkle_function_call_pre_and_post_condition(func: nip.Function,
                                                   node_name: source.NodeName,
                                                   node: source.NodeCall[source.ProgVarName | nip.GuardVarName],
+                                                  new_variables: Set[source.ExprVarT[GhostVarName]],
                                                   signatures: Mapping[str, TemporaryFunctionSignature]) -> Iterable[Insertion]:
 
     # the parameters are the "variable" in a method definition
     # the arguments are the values you pass at function call
     # (you define the parameters, you make the arguments)
     params = signatures[node.fname].parameters
     assert len(node.args) == len(params)
-    call_stash_updates = tuple(source.Update(source.ExprVar(param.typ, CallArgVarName(param.name + '/call-arg')), arg)
-                               for param, arg in zip(params, node.args))
+    call_stash_updates: Tuple[source.Update[source.ProgVarName |
```

Comment:
use a list and then freeze it to a tuple rather than concatenating tuples over and over again